hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -359,10 +359,10 @@ private Account extractAndValidateReceiver(
             } else {
                 if (!OK.equals(isUsableContract)) {
                     // call to non-existing contract flow
-                    validateTrue(!EVM_VERSION_0_30.equals(properties.evmVersion()), isUsableContract);
-                    validateTrue(!EVM_VERSION_0_34.equals(properties.evmVersion()), isUsableContract);
-                    validateTrue(!EVM_VERSION_0_38.equals(properties.evmVersion()), isUsableContract);
-                    validateTrue(properties.allowCallsToNonContractAccounts(), isUsableContract);
+                    validateTrue(isTokenAccount || !EVM_VERSION_0_30.equals(properties.evmVersion()), isUsableContract);
+                    validateTrue(isTokenAccount || !EVM_VERSION_0_34.equals(properties.evmVersion()), isUsableContract);
+                    validateTrue(isTokenAccount || !EVM_VERSION_0_38.equals(properties.evmVersion()), isUsableContract);
+                    validateTrue(isTokenAccount || properties.allowCallsToNonContractAccounts(), isUsableContract);
```

Comment:
Is the "misbehavior" mentioned in this PR's title the fact that the flag doesn't behave correctly in mono service, or that it doesn't behave correctly in mod service (or both)? I'm surprised to see changes to mono service code _and_ mod service in the same PR.

Reply:
Initially we thought there was only an issue with the mono codebase. 
But then we opened https://github.com/hashgraph/hedera-services/pull/11245 which is based on this branch to see if this PR resolves the issue and we identified issues with mod codebase as well.

Reply:
Ok, thanks!